src/transformers/generation_utils.py

```diff
@@ -841,6 +841,16 @@ def compute_transition_beam_scores(
 
         return transition_scores
 
+    def _validate_model_class(self):
+        """Confirms that the model class is compatible with generation."""
+        if not hasattr(self, "prepare_inputs_for_generation"):
+            model_class = self.__class__.__name__
+            raise TypeError(
+                f"The current model class ({model_class}) is not compatible with `.generate()`, as it doesn't have a ",
```

Comment:
(nit) Wondering if it could make sense to make a really nifty error message here where we check if any class of the model type of the given class in in one of the auto classes below - maybe over-engineered though